frontend/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,9 +1,32 @@
-import React from "react";
-import "./HomePage.css";
+import React, { useEffect, useState } from "react";
 import "bootstrap/dist/css/bootstrap.min.css";
+import { useNavigate } from "react-router-dom";
+import { QuestionEditor } from "../../components/QuestionEditor/QuestionEditor.component";
+import { SolvedTable } from "../../components/SolvedTable/SolvedTable.component";
+import { fetchUserId } from "../../api/user"; 
 
 function HomePage() {
-  return <></>;
+  const [userId, setUserId] = useState<number | null>(null);
+
+  useEffect(() => {
+    // Fetch the user ID dynamically using fetchUserId
+    fetchUserId()
+      .then((id) => {
+        if (id !== null) {
+          setUserId(id);
+        }
+      })
+      .catch((error) => {
+        console.error("Error fetching user ID: ", error);
+      });
+  }, []);
```

Comment:
are you trying to selectively render `SolvedTable` only when the user is logged in?

we can check if the user is loggedIn by using the `selectUser` selector from authSlice